modules/ldap_groups/app/models/ldap_groups/synchronized_group.rb

```diff
@@ -94,9 +97,12 @@ def add_members_to_group(new_users)
     end
 
     def remove_members_from_group(user_ids)
+      # Ensure we use pluck to get the current DB version of user_ids
+      current_user_ids = group.group_users.pluck(:user_id)
+
       call = Groups::UpdateService
         .new(user: User.system, model: group)
-        .call(user_ids: group.user_ids - user_ids)
+        .call(user_ids: current_user_ids - user_ids)
```

Comment:
This line caused the bug, we used `group.user_ids` which is cached IF it was used above in removing a user from the group.